src/Compilers/CSharp/Test/Semantic/Semantics/NullCheckedParameterTests.cs

```diff
@@ -407,13 +454,86 @@ class C
 {
     public void M()
     {
-        Func<string, string> func1 = x!=> x;
+        Func<string, string> func1 = x!!=> x;
+    }
+}";
+            CreateCompilation(source, parseOptions: TestOptions.RegularPreview).VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void NullCheckedLambdaBadSyntax()
+        {
+            var source = @"
+using System;
+class C
+{
+    public void M()
+    {
+        Func<string, string> func0 = x!=> x;
+        Func<string, string> func1 = x !=> x;
+        Func<string, string> func2 = x! => x;
+        Func<string, string> func3 = x ! => x;
```

Comment:
I would also add a case for `x != >` and `x !!= >`

Reply:
Cases added!